### PR TITLE
Feat crud emoji api

### PR DESCRIPTION
**Pull request by sadohpv**

### CODE REVIEW

**Comment by anbuihoang** [comment]
Done

File: libs/store/src/lib/settingEmoji/settingEmoji.slice.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	error: null,
+	listEmoji: {
+		staticEmoji: [],
+		animatedEmoji: [],
+	},
+};
+
+export const emojiAdapter = createEntityAdapter<StickersEntity>();
+
+export const fetchEmojisByClanId = createAsyncThunk('settingClanEmoji/fetchClanEmoji', async (clanId: string, thunkAPI) => {
```
Comment (nguyentrannhan) [inline libs/store/src/lib/settingEmoji/settingEmoji.slice.ts]: add cached

File: libs/store/src/lib/settingEmoji/settingEmoji.slice.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+export interface SettingClanEmojiState {
+	loadingStatus: LoadingStatus;
+	error?: string | null;
+	listEmoji: {
+		staticEmoji: Array<ApiClanEmojiListResponse>;
+		animatedEmoji: Array<ApiClanEmojiListResponse>;
+	};
+}
+
+type fetchEmojiRequest = {
```
Comment (nguyentrannhan) [inline libs/store/src/lib/settingEmoji/settingEmoji.slice.ts]: Type thi viet hoa chu dau

File: libs/store/src/lib/settingEmoji/settingEmoji.slice.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+			return thunkAPI.rejectWithValue({});
+		}
+	},
+);
+
+export const updateEmoji = createAsyncThunk('settingClanEmoji/updateEmoji', async ({ request, emojiId }: updateEmojiRequest, thunkAPI) => {
+	try {
+		const mezon = await ensureSession(getMezonCtx(thunkAPI));
+		const res = await mezon.client.updateClanEmojiById(mezon.session, emojiId, request);
+		if (res) {
```
Comment (nguyentrannhan) [inline libs/store/src/lib/settingEmoji/settingEmoji.slice.ts:79]: update

File: libs/store/src/lib/store.tsx [hunk 4]
```diff
@@ -125,24 +127,24 @@ export type RootState = ReturnType<typeof storeInstance.getState>;
...
+    middleware: (getDefaultMiddleware) =>
+      getDefaultMiddleware({
+        thunk: {
+          extraArgument: {
+            mezon,
+          },
+        },
+        immutableCheck: false,
+        serializableCheck: false,
+      }).prepend(errorListenerMiddleware.middleware, toastListenerMiddleware.middleware),
```
Comment (nguyentrannhan) [inline libs/store/src/lib/store.tsx]: check convension

File: libs/transport/src/lib/minio/index.ts [hunk 5]
```diff
@@ -16,12 +16,15 @@ export function uploadImageToMinIO(url: string, stream: Buffer, size: number) {
...
 }
 
-export async function handleUploadFile(client: Client, session: Session, currentClanId: string, currentChannelId: string, filename: string, file: File, mode?: number): Promise<ApiMessageAttachment> {
+export async function handleUploadFile(client: Client, session: Session, currentClanId: string, currentChannelId: string, filename: string, file: File, mode?: number, path?: string): Promise<ApiMessageAttachment> {
 	
 	return new Promise<ApiMessageAttachment>(async function (resolve, reject) {
 		try {
 			const ms = (new Date()).getTime();
 			let fullfilename;
+      if(mode === 3) {
```
Comment (nguyentrannhan) [inline libs/transport/src/lib/minio/index.ts]: dung enum channel mode

File: libs/utils/src/lib/utils/index.ts [hunk 6]
```diff
@@ -296,6 +296,16 @@ export const formatTimeToMMSS = (duration: number): string => {
   return `${minutes}:${seconds < 10 ? '0' : ''}${seconds}`;
 };
 
+export const isImageFile = (fileSrc: string) => {
```
Comment (nguyentrannhan) [inline libs/utils/src/lib/utils/index.ts]: ko can tach ra 2 phan

File: libs/transport/src/lib/minio/index.ts [hunk 7]
```diff
@@ -16,12 +16,15 @@ export function uploadImageToMinIO(url: string, stream: Buffer, size: number) {
...
+        fullfilename = path + '/' + currentClanId + '/' + currentChannelId + '/' + ms + filename.replace(/-|\(|\)| /g,"_")
+      } else
 			if(mode === 2){
 				fullfilename = currentClanId + '/' + currentChannelId + '/' + ms + filename.replace(/-|\(|\)| /g,"_");
 			}else{
```
Comment (nguyentrannhan) [inline libs/transport/src/lib/minio/index.ts]: remove mode === 3   

thay mode === 2 bang enum

neu path ko rong, fullfilenam = path + '/' + fullfilename

trong else {}, them directid sau currentChannelId + '/'

**Pull request merged**